storage/http_client.go

```diff
@@ -935,6 +935,80 @@ func (c *httpStorageClient) DeleteHMACKey(ctx context.Context, desc *hmacKeyDesc
 	return errMethodNotSupported
 }
 
+// Notification methods.
+
+func (c *httpStorageClient) ListNotifications(ctx context.Context, bucket string, opts ...storageOption) (n map[string]*Notification, err error) {
+	ctx = trace.StartSpan(ctx, "cloud.google.com/go/storage.httpStorageClient.ListNotifications")
+	defer func() { trace.EndSpan(ctx, err) }()
+
+	s := callSettings(c.settings, opts...)
+	call := c.raw.Notifications.List(bucket)
+	if s.userProject != "" {
+		call.UserProject(s.userProject)
+	}
+	var res *raw.Notifications
+	err = run(ctx, func() error {
+		res, err = call.Context(ctx).Do()
+		return err
+	}, s.retry, true, setRetryHeaderHTTP(call))
+	if err != nil {
+		return nil, err
+	}
+	return notificationsToMap(res.Items), nil
```

Comment:
Interestingly it seems like this implementation doesn't use page tokens-- so if there are more than 1000 notifications you won't get all of them. (I think this would be very uncommon though)

Reply:
Might be worth adding a Note or TODO for later mentioning this. I too was surprised to see an un-paginated list api, even if it predates the design standards.

Reply:
Ah so I did a bit of digging and found the following:
* The JSON API [doesn't actually support page tokens](https://cloud.google.com/storage/docs/json_api/v1/notifications/list), probably because...
* There is a [quota](https://cloud.google.com/storage/quotas#buckets) of 100 notifications per bucket.

I'm not sure how this would effect gRPC, if at all.

Reply:
I think we should keep page size unspecified. The server will "do the right thing" and if for some reason that quota is raised to 500, we won't have to change our code or be worried about what might happen - it will "just work".

Can we add a comment that the server will use a default page_size (just sent an email to inquire what the default actually is), and that our code will not set a page_size? We could leave the param just in case we want to set it later. WDYT?

Reply:
Sounds fine to me! Agreed that we don't want to depend of behavior that may change going forward.

Reply:
Gotcha... IIUC, based on the [same block of code](https://github.com/googleapis/google-cloud-go/blob/main/storage/internal/apiv2/storage_client.go#L704-L725), `req.GetPageSize()` is not set here and the zero value fallbacks to the API default pageSize of 100. 

So I've added a comment but no change in our code. PTAL and let me know if my understanding is correct.